nexus/db-queries/src/db/datastore/console_session.rs

```diff
@@ -13,13 +13,49 @@ use async_bb8_diesel::AsyncRunQueryDsl;
 use chrono::Utc;
 use diesel::prelude::*;
 use nexus_db_lookup::LookupPath;
+use nexus_db_schema::schema::console_session;
 use omicron_common::api::external::CreateResult;
 use omicron_common::api::external::DeleteResult;
 use omicron_common::api::external::Error;
 use omicron_common::api::external::InternalContext;
+use omicron_common::api::external::LookupResult;
+use omicron_common::api::external::LookupType;
+use omicron_common::api::external::ResourceType;
 use omicron_common::api::external::UpdateResult;
+use omicron_uuid_kinds::GenericUuid;
 
 impl DataStore {
+    pub async fn session_lookup_by_token(
```

Comment:
maybe worth a doc comment explaining what this does and why it differs from the "normal" lookup machinery? not a big deal.